core/src/main/java/com/digitalpebble/stormcrawler/protocol/httpclient/HttpProtocol.java

```diff
@@ -99,24 +107,42 @@ public void configure(final Config conf) {
         String proxyHost = ConfUtils.getString(conf, "http.proxy.host", null);
         int proxyPort = ConfUtils.getInt(conf, "http.proxy.port", 8080);
 
+        String proxyUser = ConfUtils.getString(conf, "http.proxy.user", null);
+        String proxyPass = ConfUtils.getString(conf, "http.proxy.pass", null);
+        String proxyAuthScheme = ConfUtils.getString(conf, "http.proxy.authscheme", AuthSchemes.BASIC);
+        List<String> authSchemes = Lists.newArrayList();
+
         boolean useProxy = proxyHost != null && proxyHost.length() > 0;
 
         // use a proxy?
         if (useProxy) {
+
+            if (StringUtils.isNotBlank(proxyUser) && StringUtils.isNotBlank(proxyPass)) {
+                if (!SUPPORTED_AUTH_SCHEMES.contains(proxyAuthScheme)) {
+                    LOG.error("Configured proxy auth scheme: {} is not supported. Proxy authentication will not be used", proxyAuthScheme);
+                } else {
+                    authSchemes.add(proxyAuthScheme);
+                    BasicCredentialsProvider basicAuthCreds = new BasicCredentialsProvider();
+                    basicAuthCreds.setCredentials(new AuthScope(proxyHost, proxyPort),
+                            new UsernamePasswordCredentials(proxyUser, proxyPass));
+                    builder.setDefaultCredentialsProvider(basicAuthCreds);
+                }
+            }
+
             HttpHost proxy = new HttpHost(proxyHost, proxyPort);
-            DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(
-                    proxy);
+            DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(proxy);
```

Comment:
I'm happy to change for longer lines (or shorter ones) What matters is that we don't get stuff in diffs that are due to formatting discrepancies